sonar-scanner-engine/src/main/java/org/sonar/scanner/report/MeasuresPublisher.java

```diff
@@ -100,47 +117,76 @@ private static void setValueAccordingToType(ScannerReport.Measure.Builder builde
     this.measureCache = measureCache;
+    this.testPlanBuilder = testPlanBuilder;
   }
 
   @Override
   public void publish(ScannerReportWriter writer) {
     for (final BatchComponent component : componentCache.all()) {
       // Recompute all coverage measures from line data to take into account the possible merge of several reports
-      DefaultMeasure<String> lineHitsMeasure = (DefaultMeasure<String>) measureCache.byMetric(component.key(), CoreMetrics.COVERAGE_LINE_HITS_DATA_KEY);
-      if (lineHitsMeasure != null) {
-        Map<Integer, Integer> lineHits = KeyValueFormat.parseIntInt(lineHitsMeasure.value());
-        measureCache.put(component.key(), LINES_TO_COVER_KEY, new DefaultMeasure<Integer>().forMetric(LINES_TO_COVER).withValue(lineHits.keySet().size()));
-        measureCache.put(component.key(), UNCOVERED_LINES_KEY,
-          new DefaultMeasure<Integer>().forMetric(UNCOVERED_LINES).withValue((int) lineHits.values()
-            .stream()
-            .filter(hit -> hit == 0)
-            .count()));
-      }
-      DefaultMeasure<String> conditionsMeasure = (DefaultMeasure<String>) measureCache.byMetric(component.key(), CoreMetrics.CONDITIONS_BY_LINE_KEY);
-      DefaultMeasure<String> coveredConditionsMeasure = (DefaultMeasure<String>) measureCache.byMetric(component.key(), CoreMetrics.COVERED_CONDITIONS_BY_LINE_KEY);
-      if (conditionsMeasure != null) {
-        Map<Integer, Integer> conditions = KeyValueFormat.parseIntInt(conditionsMeasure.value());
-        Map<Integer, Integer> coveredConditions = coveredConditionsMeasure != null ? KeyValueFormat.parseIntInt(coveredConditionsMeasure.value()) : Collections.emptyMap();
-        measureCache.put(component.key(), CONDITIONS_TO_COVER_KEY, new DefaultMeasure<Integer>().forMetric(CONDITIONS_TO_COVER).withValue(conditions
-          .values()
-          .stream()
-          .mapToInt(Integer::intValue)
-          .sum()));
-        measureCache.put(component.key(), UNCOVERED_CONDITIONS_KEY,
-          new DefaultMeasure<Integer>().forMetric(UNCOVERED_CONDITIONS)
-            .withValue((int) conditions.keySet()
-              .stream()
-              .mapToInt(line -> conditions.get(line) - coveredConditions.get(line))
-              .sum()));
-      }
+      updateCoverageFromLineData(component);
+      // Recompute test execution measures from MutableTestPlan to take into account the possible merge of several reports
+      updateTestExecutionFromTestPlan(component);
+
       Iterable<DefaultMeasure<?>> scannerMeasures = measureCache.byComponentKey(component.key());
       Iterable<ScannerReport.Measure> reportMeasures = transform(scannerMeasures, new MeasureToReportMeasure(component));
       writer.writeComponentMeasures(component.batchId(), reportMeasures);
     }
   }
 
+  private void updateTestExecutionFromTestPlan(final BatchComponent component) {
+    final MutableTestPlan testPlan = testPlanBuilder.loadPerspective(MutableTestPlan.class, component.inputComponent());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "MutableTestPlan"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "MutableTestPlan"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)